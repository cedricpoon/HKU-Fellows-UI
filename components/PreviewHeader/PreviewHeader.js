import React, { Component } from 'react';
import PropTypes from "prop-types";
import { Header, Item, Input, Icon, Button, Right, Text } from 'native-base';
import { Keyboard } from 'react-native';

import NavigationService from 'hkufui/src/NavigationService';
import { displayName as appName } from 'hkufui/app.json';
import { localize } from 'hkufui/locale';

const locale = localize({ language: 'en', country: 'hk' });

import styles from './Styles'

class PreviewHeader extends Component {
  constructor(props) {
    super(props);
    this.state = { inputFocused: false, query: '' };
  }

  searchBarOnFocus = () => {
    this.setState(() => ({
      inputFocused: true
    }));
  }

  searchCancel = () => {
    this._searchBar._root.blur();
    this._searchBar._root.clear();

    Keyboard.dismiss();

    this.props.onCancel(this.state.query !== '');

    this.setState(() => ({
      inputFocused: false,
      query: ''
    }));
  }

  render() {
    const { location, onSearchThunk } = this.props;
    const { inputFocused, query } = this.state;

    const inputIcon = location === '' ? 'ios-arrow-dropright' : 'search'

    return (
      <Header searchBar style={styles.header}>
        <Item rounded>
          <Button
            rounded
            style={styles.leftLabel}
            onPress={() => {
              NavigationService.navigate('SelectCourse');
            }}
          >
            <Text style={styles.leftLabelText}>
              {location !== '' ? location : appName}
            </Text>
          </Button>
          <Input
            placeholder={
              location !== '' ?
                `${locale['header.searchPlaceholder']}`
              :
                `${locale['header.noCoursePlaceholder']}`
            }
            style={styles.input}
            returnKeyType="search"
            onFocus={this.searchBarOnFocus}
            ref={(ref) => { this._searchBar = ref }}
            enablesReturnKeyAutomatically
            autoCorrect={false}
            onSubmitEditing={onSearchThunk(query)}
            onChangeText={(query) => this.setState({query})}
            disabled={location === ''}
          />
          <Right style={styles.rightButtons}>
            <Button
              transparent
              small
              dark
              disabled={!inputFocused}
              onPress={this.searchCancel}
            >
              <Icon name={inputFocused ? "close-circle-outline" : inputIcon} style={styles.rightIcon} />
            </Button>
          </Right>

        </Item>
      </Header>
    );
  }
}

PreviewHeader.defaultProps = {
  onSearchThunk: () => {},
  onCancel: () => {}
}

PreviewHeader.propTypes = {
  location: PropTypes.string.isRequired,
  onSearchThunk: PropTypes.func,
  onCancel: PropTypes.func
};

export default PreviewHeader;
