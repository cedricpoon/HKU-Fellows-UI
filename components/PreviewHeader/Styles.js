import { StyleSheet } from 'react-native';

import { header, post } from 'hkufui/theme/grid';
import { logo } from 'hkufui/theme/palette';
import nbColor from 'hkufui/native-base-theme/variables/commonColor'

export default StyleSheet.create({
  header: {
    backgroundColor: nbColor.toolbarDefaultBg
  },
  leftLabel: {
    alignSelf: 'center',
    height: null,
    backgroundColor: logo.black,
    marginLeft: header.labelMarginLeft
  },
  leftLabelText: {
    paddingLeft: header.labelPadding,
    paddingRight: header.labelPadding,
    fontSize: post.miniSize
  },
  input: {
    fontSize: post.titleSize,
    top: 0
  },
  rightButtons: {
    flex: 0
  },
  rightIcon: {
    marginLeft: 0,
    marginRight: header.iconMarginRight
  }
});
