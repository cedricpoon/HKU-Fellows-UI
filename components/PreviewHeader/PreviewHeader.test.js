import React from 'react';
import { shallow } from 'enzyme';
import PreviewHeader from './PreviewHeader';

describe('Testing PreviewHeader Component', () => {
  it('renders as expected with required props', () => {
    const wrapper = shallow(
      <PreviewHeader location='' />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('will search', () => {
    const wrapper = shallow(
      <PreviewHeader location='' onSearchThunk={() => {}} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('will cancel', () => {
    const wrapper = shallow(
      <PreviewHeader location='' onCancel={() => {}} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
