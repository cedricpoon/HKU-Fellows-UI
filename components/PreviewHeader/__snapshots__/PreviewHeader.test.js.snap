// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing PreviewHeader Component renders as expected with required props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PreviewHeader
    location=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Styled(Item)
        rounded={true}
      >
        <Styled(Button)
          onPress={[Function]}
          rounded={true}
          style={
            Object {
              "alignSelf": "center",
              "backgroundColor": "#4E4A47",
              "height": null,
              "marginLeft": 2.5,
            }
          }
        >
          <Styled(Text)
            style={
              Object {
                "fontSize": 10,
                "paddingLeft": 10,
                "paddingRight": 10,
              }
            }
          >
            HKU Fellows
          </Styled(Text)>
        </Styled(Button)>
        <Styled(Input)
          disabled={true}
          enablesReturnKeyAutomatically={true}
          onFocus={[Function]}
          placeholder="Swipe left to choose course"
          returnKeyType="search"
          style={
            Object {
              "fontSize": 15,
              "top": 0,
            }
          }
        />
        <Styled(Right)
          style={
            Object {
              "flex": 0,
            }
          }
        >
          <Styled(Button)
            dark={true}
            disabled={true}
            onPress={[Function]}
            small={true}
            transparent={true}
          >
            <Styled(Icon)
              name="arrow-dropright"
              style={
                Object {
                  "marginLeft": 0,
                  "marginRight": 15,
                }
              }
            />
          </Styled(Button)>
        </Styled(Right)>
      </Styled(Item)>,
      "searchBar": true,
      "style": Object {
        "backgroundColor": "#F8F8F8",
      },
      "virtual": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Styled(Button)
            onPress={[Function]}
            rounded={true}
            style={
              Object {
                "alignSelf": "center",
                "backgroundColor": "#4E4A47",
                "height": null,
                "marginLeft": 2.5,
              }
            }
          >
            <Styled(Text)
              style={
                Object {
                  "fontSize": 10,
                  "paddingLeft": 10,
                  "paddingRight": 10,
                }
              }
            >
              HKU Fellows
            </Styled(Text)>
          </Styled(Button)>,
          <Styled(Input)
            disabled={true}
            enablesReturnKeyAutomatically={true}
            onFocus={[Function]}
            placeholder="Swipe left to choose course"
            returnKeyType="search"
            style={
              Object {
                "fontSize": 15,
                "top": 0,
              }
            }
          />,
          <Styled(Right)
            style={
              Object {
                "flex": 0,
              }
            }
          >
            <Styled(Button)
              dark={true}
              disabled={true}
              onPress={[Function]}
              small={true}
              transparent={true}
            >
              <Styled(Icon)
                name="arrow-dropright"
                style={
                  Object {
                    "marginLeft": 0,
                    "marginRight": 15,
                  }
                }
              />
            </Styled(Button)>
          </Styled(Right)>,
        ],
        "rounded": true,
        "virtual": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Styled(Text)
              style={
                Object {
                  "fontSize": 10,
                  "paddingLeft": 10,
                  "paddingRight": 10,
                }
              }
            >
              HKU Fellows
            </Styled(Text)>,
            "onPress": [Function],
            "rounded": true,
            "style": Object {
              "alignSelf": "center",
              "backgroundColor": "#4E4A47",
              "height": null,
              "marginLeft": 2.5,
            },
            "virtual": undefined,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "HKU Fellows",
              "style": Object {
                "fontSize": 10,
                "paddingLeft": 10,
                "paddingRight": 10,
              },
              "virtual": undefined,
            },
            "ref": null,
            "rendered": "HKU Fellows",
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "disabled": true,
            "enablesReturnKeyAutomatically": true,
            "onFocus": [Function],
            "placeholder": "Swipe left to choose course",
            "returnKeyType": "search",
            "style": Object {
              "fontSize": 15,
              "top": 0,
            },
            "virtual": undefined,
          },
          "ref": [Function],
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Styled(Button)
              dark={true}
              disabled={true}
              onPress={[Function]}
              small={true}
              transparent={true}
            >
              <Styled(Icon)
                name="arrow-dropright"
                style={
                  Object {
                    "marginLeft": 0,
                    "marginRight": 15,
                  }
                }
              />
            </Styled(Button)>,
            "style": Object {
              "flex": 0,
            },
            "virtual": undefined,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Styled(Icon)
                name="arrow-dropright"
                style={
                  Object {
                    "marginLeft": 0,
                    "marginRight": 15,
                  }
                }
              />,
              "dark": true,
              "disabled": true,
              "onPress": [Function],
              "small": true,
              "transparent": true,
              "virtual": undefined,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "name": "arrow-dropright",
                "style": Object {
                  "marginLeft": 0,
                  "marginRight": 15,
                },
                "virtual": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Styled(Item)
          rounded={true}
        >
          <Styled(Button)
            onPress={[Function]}
            rounded={true}
            style={
              Object {
                "alignSelf": "center",
                "backgroundColor": "#4E4A47",
                "height": null,
                "marginLeft": 2.5,
              }
            }
          >
            <Styled(Text)
              style={
                Object {
                  "fontSize": 10,
                  "paddingLeft": 10,
                  "paddingRight": 10,
                }
              }
            >
              HKU Fellows
            </Styled(Text)>
          </Styled(Button)>
          <Styled(Input)
            disabled={true}
            enablesReturnKeyAutomatically={true}
            onFocus={[Function]}
            placeholder="Swipe left to choose course"
            returnKeyType="search"
            style={
              Object {
                "fontSize": 15,
                "top": 0,
              }
            }
          />
          <Styled(Right)
            style={
              Object {
                "flex": 0,
              }
            }
          >
            <Styled(Button)
              dark={true}
              disabled={true}
              onPress={[Function]}
              small={true}
              transparent={true}
            >
              <Styled(Icon)
                name="arrow-dropright"
                style={
                  Object {
                    "marginLeft": 0,
                    "marginRight": 15,
                  }
                }
              />
            </Styled(Button)>
          </Styled(Right)>
        </Styled(Item)>,
        "searchBar": true,
        "style": Object {
          "backgroundColor": "#F8F8F8",
        },
        "virtual": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Styled(Button)
              onPress={[Function]}
              rounded={true}
              style={
                Object {
                  "alignSelf": "center",
                  "backgroundColor": "#4E4A47",
                  "height": null,
                  "marginLeft": 2.5,
                }
              }
            >
              <Styled(Text)
                style={
                  Object {
                    "fontSize": 10,
                    "paddingLeft": 10,
                    "paddingRight": 10,
                  }
                }
              >
                HKU Fellows
              </Styled(Text)>
            </Styled(Button)>,
            <Styled(Input)
              disabled={true}
              enablesReturnKeyAutomatically={true}
              onFocus={[Function]}
              placeholder="Swipe left to choose course"
              returnKeyType="search"
              style={
                Object {
                  "fontSize": 15,
                  "top": 0,
                }
              }
            />,
            <Styled(Right)
              style={
                Object {
                  "flex": 0,
                }
              }
            >
              <Styled(Button)
                dark={true}
                disabled={true}
                onPress={[Function]}
                small={true}
                transparent={true}
              >
                <Styled(Icon)
                  name="arrow-dropright"
                  style={
                    Object {
                      "marginLeft": 0,
                      "marginRight": 15,
                    }
                  }
                />
              </Styled(Button)>
            </Styled(Right)>,
          ],
          "rounded": true,
          "virtual": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Styled(Text)
                style={
                  Object {
                    "fontSize": 10,
                    "paddingLeft": 10,
                    "paddingRight": 10,
                  }
                }
              >
                HKU Fellows
              </Styled(Text)>,
              "onPress": [Function],
              "rounded": true,
              "style": Object {
                "alignSelf": "center",
                "backgroundColor": "#4E4A47",
                "height": null,
                "marginLeft": 2.5,
              },
              "virtual": undefined,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "HKU Fellows",
                "style": Object {
                  "fontSize": 10,
                  "paddingLeft": 10,
                  "paddingRight": 10,
                },
                "virtual": undefined,
              },
              "ref": null,
              "rendered": "HKU Fellows",
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "disabled": true,
              "enablesReturnKeyAutomatically": true,
              "onFocus": [Function],
              "placeholder": "Swipe left to choose course",
              "returnKeyType": "search",
              "style": Object {
                "fontSize": 15,
                "top": 0,
              },
              "virtual": undefined,
            },
            "ref": [Function],
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Styled(Button)
                dark={true}
                disabled={true}
                onPress={[Function]}
                small={true}
                transparent={true}
              >
                <Styled(Icon)
                  name="arrow-dropright"
                  style={
                    Object {
                      "marginLeft": 0,
                      "marginRight": 15,
                    }
                  }
                />
              </Styled(Button)>,
              "style": Object {
                "flex": 0,
              },
              "virtual": undefined,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Styled(Icon)
                  name="arrow-dropright"
                  style={
                    Object {
                      "marginLeft": 0,
                      "marginRight": 15,
                    }
                  }
                />,
                "dark": true,
                "disabled": true,
                "onPress": [Function],
                "small": true,
                "transparent": true,
                "virtual": undefined,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "name": "arrow-dropright",
                  "style": Object {
                    "marginLeft": 0,
                    "marginRight": 15,
                  },
                  "virtual": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
